### PR TITLE
Cleanup Maven POMs

### LINKED ISSUE #400: Re-enable checks on connectors project
See comments in https://github.com/syndesisio/syndesis/pull/365, we should enable checks done by Maven plugins that are currently disabled by setting the `phase` to `none` or `skip` set to `true`. That means cleaning up the code to meet the standards of those checks also.

### PR DESCRIPTION
**Pull request by zregvart**
This addresses the overall tech debt that we accumulated in Maven POMs. I tried to make sure that everything still works properly though with such big refactorings there is always a risk of breaking something.

Please have a look and see if you see something obviously wrong.

It's still a work in progress, there are some issues with `server/cli` that I'm trying to fix.

I've checked the difference between the last daily release in the `server` fat JAR and I've included them here. The most jarring difference is the kubernetes/openshift client/model change. I think it makes sense to be on a newer version but we had previously encountered issues with those.

I'm also unsure about zipkin dependencies, seems that 2.2.1 and 2.2.2 have different groupIds, but they are both declared in the Spring Boot Zipkin starter.

I'll also try to perform some manual tests to double check.

@syndesisio/backend mind having a look?

```diff
--- <unnamed>
+++ <unnamed>
@@ -5,7 +5,6 @@
 /lib/archaius-core-0.7.4.jar
 /lib/aspectjweaver-1.8.13.jar
 /lib/atlas-api-1.35.0.jar
-/lib/atlas-core-1.35.0-tests.jar
 /lib/atlas-core-1.35.0.jar
 /lib/atlas-java-core-1.35.0.jar
 /lib/atlas-java-model-1.35.0.jar
@@ -23,31 +22,29 @@
 /lib/atlas-xml-service-1.35.0.jar
 /lib/automaton-1.11-8.jar
 /lib/base64-2.3.8.jar
-/lib/bcpkix-jdk15on-1.55.jar
-/lib/bcprov-jdk15on-1.55.jar
+/lib/bcpkix-jdk15on-1.56.jar
+/lib/bcprov-jdk15on-1.56.jar
 /lib/btf-1.2.jar
 /lib/classmate-1.3.4.jar
 /lib/common-model-1.4-SNAPSHOT.jar
 /lib/common-resteasy-spring-boot-starter-1.4-SNAPSHOT.jar
-/lib/common-util-1.4-SNAPSHOT.jar
 /lib/commons-codec-1.10.jar
 /lib/commons-compress-1.14.jar
 /lib/commons-configuration-1.8.jar
 /lib/commons-io-2.5.jar
 /lib/commons-lang-2.6.jar
 /lib/commons-lang3-3.2.1.jar
-/lib/commons-logging-1.2.jar
+/lib/common-util-1.4-SNAPSHOT.jar
 /lib/compiler-0.9.2.jar
 /lib/connector-support-catalog-1.4-SNAPSHOT.jar
 /lib/dom4j-1.6.1.jar
-/lib/error_prone_annotations-2.0.19.jar
+/lib/error_prone_annotations-2.3.1.jar
 /lib/extension-converter-1.4-SNAPSHOT.jar
-/lib/findbugs-annotations-3.0.1.jar
 /lib/generex-1.0.1.jar
 /lib/guava-23.3-jre.jar
-/lib/h2-1.4.193.jar
+/lib/h2-1.4.197.jar
 /lib/hibernate-jpa-2.1-api-1.0.0.Final.jar
-/lib/hibernate-validator-5.3.5.Final.jar
+/lib/hibernate-validator-5.3.6.Final.jar
 /lib/httpclient-4.5.5.jar
 /lib/httpcore-4.4.9.jar
 /lib/hystrix-core-1.5.12.jar
@@ -55,16 +52,16 @@
 /lib/integration-project-generator-1.4-SNAPSHOT.jar
 /lib/j2objc-annotations-1.1.jar
 /lib/jackson-annotations-2.8.10.jar
-/lib/jackson-core-2.8.10.jar
+/lib/jackson-core-2.8.11.jar
 /lib/jackson-coreutils-1.6.jar
 /lib/jackson-databind-2.8.10.jar
-/lib/jackson-dataformat-xml-2.8.10.jar
-/lib/jackson-dataformat-yaml-2.8.10.jar
+/lib/jackson-dataformat-xml-2.8.11.jar
+/lib/jackson-dataformat-yaml-2.8.11.jar
 /lib/jackson-datatype-jdk8-2.8.10.jar
-/lib/jackson-jaxrs-base-2.8.10.jar
-/lib/jackson-jaxrs-json-provider-2.8.10.jar
+/lib/jackson-jaxrs-base-2.8.11.jar
+/lib/jackson-jaxrs-json-provider-2.8.11.jar
 /lib/jackson-json-reference-core-0.2.1.jar
-/lib/jackson-module-jaxb-annotations-2.8.10.jar
+/lib/jackson-module-jaxb-annotations-2.8.11.jar
 /lib/jackson-module-jsonSchema-2.8.10.jar
 /lib/javassist-3.21.0-GA.jar
 /lib/javax.el-3.0.0.jar
@@ -75,7 +72,7 @@
 /lib/javax.ws.rs-api-2.1.jar
 /lib/jaxb-core-2.3.0.jar
 /lib/jaxb-impl-2.3.0.jar
-/lib/jboss-annotations-api_1.2_spec-1.0.0.Final.jar
+/lib/jboss-annotations-api_1.2_spec-1.0.2.Final.jar
 /lib/jboss-jaxb-api_2.3_spec-1.0.0.Final.jar
 /lib/jboss-jaxrs-api_2.1_spec-1.0.0.Final.jar
 /lib/jboss-logging-3.3.2.Final.jar
@@ -87,7 +84,6 @@
 /lib/jdbi-2.78.jar
 /lib/joda-time-2.9.9.jar
 /lib/jolokia-client-java-1.4.0.jar
-/lib/joor-0.9.4.jar
 /lib/jopt-simple-5.0.3.jar
 /lib/json-patch-1.9.jar
 /lib/json-schema-core-1.2.8.jar
@@ -95,19 +91,19 @@
 /lib/json-simple-1.1.1.jar
 /lib/jsr305-3.0.1.jar
 /lib/jul-to-slf4j-1.7.25.jar
-/lib/kubernetes-client-2.4.1.jar
-/lib/kubernetes-model-1.0.74.jar
+/lib/kubernetes-client-3.1.4.jar
+/lib/kubernetes-model-2.0.4.jar
 /lib/libphonenumber-8.0.0.jar
 /lib/log4j-over-slf4j-1.7.25.jar
 /lib/logback-classic-1.1.11.jar
 /lib/logback-core-1.1.11.jar
-/lib/logging-interceptor-3.8.0.jar
+/lib/logging-interceptor-3.8.1.jar
 /lib/microprofile-config-api-1.1.jar
 /lib/microprofile-rest-client-api-1.0.jar
 /lib/msg-simple-1.1.jar
-/lib/okhttp-3.8.0.jar
+/lib/okhttp-3.8.1.jar
 /lib/okio-1.13.0.jar
-/lib/openshift-client-2.4.1.jar
+/lib/openshift-client-3.1.4.jar
 /lib/org.osgi.annotation.versioning-1.0.0.jar
 /lib/postgresql-9.4.1212.jre7.jar
 /lib/reactive-streams-1.0.2.jar
@@ -138,8 +134,10 @@
 /lib/server-update-controller-1.4-SNAPSHOT.jar
 /lib/server-verifier-1.4-SNAPSHOT.jar
 /lib/slf4j-api-1.7.25.jar
+/lib/slf4j-ext-1.7.25.jar
 /lib/snakeyaml-1.17.jar
-/lib/spring-aop-4.3.12.RELEASE.jar
+/lib/spotbugs-annotations-3.1.5.jar
+/lib/spring-aop-4.3.17.RELEASE.jar
 /lib/spring-beans-4.3.17.RELEASE.jar
 /lib/spring-boot-1.5.13.RELEASE.jar
 /lib/spring-boot-actuator-1.5.13.RELEASE.jar
@@ -154,26 +152,26 @@
 /lib/spring-boot-starter-web-1.5.13.RELEASE.jar
 /lib/spring-boot-starter-websocket-1.5.13.RELEASE.jar
 /lib/spring-boot-test-1.5.13.RELEASE.jar
-/lib/spring-cloud-commons-1.2.4.RELEASE.jar
-/lib/spring-cloud-context-1.2.4.RELEASE.jar
+/lib/spring-cloud-commons-1.2.5.RELEASE.jar
+/lib/spring-cloud-context-1.2.5.RELEASE.jar
 /lib/spring-cloud-security-1.2.1.RELEASE.jar
-/lib/spring-cloud-sleuth-core-1.2.5.RELEASE.jar
-/lib/spring-cloud-sleuth-zipkin-1.2.5.RELEASE.jar
-/lib/spring-cloud-starter-1.2.4.RELEASE.jar
+/lib/spring-cloud-sleuth-core-1.2.6.RELEASE.jar
+/lib/spring-cloud-sleuth-zipkin-1.2.6.RELEASE.jar
+/lib/spring-cloud-starter-1.2.5.RELEASE.jar
 /lib/spring-cloud-starter-security-1.2.1.RELEASE.jar
-/lib/spring-cloud-starter-sleuth-1.2.5.RELEASE.jar
-/lib/spring-cloud-starter-zipkin-1.2.5.RELEASE.jar
+/lib/spring-cloud-starter-sleuth-1.2.6.RELEASE.jar
+/lib/spring-cloud-starter-zipkin-1.2.6.RELEASE.jar
 /lib/spring-context-4.3.17.RELEASE.jar
 /lib/spring-core-4.3.12.RELEASE.jar
 /lib/spring-expression-4.3.17.RELEASE.jar
 /lib/spring-jdbc-4.3.17.RELEASE.jar
 /lib/spring-messaging-4.3.17.RELEASE.jar
-/lib/spring-security-config-4.2.3.RELEASE.jar
-/lib/spring-security-core-4.2.3.RELEASE.jar
-/lib/spring-security-crypto-4.2.3.RELEASE.jar
+/lib/spring-security-config-4.2.6.RELEASE.jar
+/lib/spring-security-core-4.2.6.RELEASE.jar
+/lib/spring-security-crypto-4.2.6.RELEASE.jar
 /lib/spring-security-rsa-1.0.3.RELEASE.jar
-/lib/spring-security-web-4.2.3.RELEASE.jar
-/lib/spring-social-config-1.1.4.RELEASE.jar
+/lib/spring-security-web-4.2.6.RELEASE.jar
+/lib/spring-social-config-1.1.6.RELEASE.jar
 /lib/spring-social-core-1.1.6.RELEASE.jar
 /lib/spring-social-salesforce-1.3.0.jar
 /lib/spring-social-twitter-1.1.2.RELEASE.jar
@@ -190,8 +188,8 @@
 /lib/swagger-parser-1.0.34.jar
 /lib/tomcat-jdbc-8.5.31.jar
 /lib/tomcat-juli-8.5.31.jar
-/lib/undertow-core-1.4.21.Final.jar
-/lib/undertow-servlet-1.4.21.Final.jar
+/lib/undertow-core-1.4.25.Final.jar
+/lib/undertow-servlet-1.4.25.Final.jar
 /lib/undertow-websockets-jsr-1.4.25.Final.jar
 /lib/uri-template-0.9.jar
 /lib/validation-api-1.1.0.Final.jar
@@ -201,6 +199,7 @@
 /lib/xnio-api-3.3.8.Final.jar
 /lib/xnio-nio-3.3.8.Final.jar
 /lib/xsom-20140925.jar
-/lib/zipkin-1.28.0.jar
-/lib/zipkin-reporter-1.0.0.jar
+/lib/zipkin-2.2.1.jar
+/lib/zipkin-2.2.2.jar
+/lib/zipkin-reporter-1.1.2.jar
 /lib/zjsonpatch-0.3.0.jar
```

Fixes #400

### CODE REVIEW

**Comment by zregvart** [comment]
@oscerd I've added Jackson upgrade in https://github.com/syndesisio/syndesis/pull/2930/commits/77f2f15e07c9298ba054b7905952b540be9a6538 I think this might be good for a first pass.

File: app/connector/ftp/pom.xml [hunk 1]
```diff
@@ -15,92 +15,93 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.camel</groupId>
+      <artifactId>camel-ftp</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>commons-net</groupId>
+      <artifactId>commons-net</artifactId>
+      <version>3.6</version>
```
Comment (rhuss) [inline app/connector/ftp/pom.xml]: Maybe externalize or move it up in the dependency list ?

File: app/server/dao/pom.xml [hunk 2]
```diff
@@ -110,16 +111,23 @@
...
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
+      <scope>test</scope>
     </dependency>
 
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
+      <scope>test</scope>
```
Comment (rhuss) [inline app/server/dao/pom.xml]: Isn't the scope set in the parent, too ?

File: app/server/endpoint/pom.xml [hunk 3]
```diff
@@ -192,13 +189,18 @@
...
     <dependency>
-      <groupId>javax.ws.rs</groupId>
-      <artifactId>javax.ws.rs-api</artifactId>
+      <groupId>org.jboss.spec.javax.annotation</groupId>
+      <artifactId>jboss-annotations-api_1.2_spec</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>org.hibernate.javax.persistence</groupId>
+      <artifactId>hibernate-jpa-2.1-api</artifactId>
```
Comment (rhuss) [inline app/server/endpoint/pom.xml]: Why do we have a hibernate-jpa dependency in the rest endpoint module ?

File: app/server/runtime/pom.xml [hunk 4]
```diff
@@ -967,8 +852,14 @@
     <dependency>
       <groupId>io.fabric8</groupId>
       <artifactId>spring-cloud-kubernetes-core</artifactId>
-      <version>${spring-cloud-kubernetes-core.version}</version>
+      <version>0.1.6</version>
```
Comment (rhuss) [inline app/server/runtime/pom.xml]: Should this be moved up ? Not sure how to deal with version numbers like this.

File: app/connector/aws-s3/pom.xml [hunk 5]
```diff
@@ -52,14 +82,33 @@
...
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-s3</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.github.spotbugs</groupId>
+      <artifactId>spotbugs-annotations</artifactId>
```
[the remark this answers is not in the document]
Comment (zregvart) [reply app/connector/aws-s3/pom.xml]: Spotbugs is the friendly fork of Findbugs caused by Findbugs not being maintained any more.

**Comment by zregvart** [comment]
@rhuss on the version numbers, I tend to use versions in place if there is only one use of that dependency, and properties if it's used multiple times. To me this makes sense as using property or mentioning the version inline if the that version is used only for a single dependency is still only one occurrence of that version.

At least I tried to use that reasoning. I'm sure I did not do this consistently, so there might be places where I failed to apply this logic.

File: app/server/endpoint/pom.xml (hunk 3 above)
Comment (zregvart) [reply app/server/endpoint/pom.xml]: I have no idea :)

**Comment by zregvart** [comment]
I'm sure that we'll have to do another pass and cleanup a bit more. I would say this is a good first iteration and we work on this in smaller increments onwards.

**Pull request merged**